src/chunk_size/rust_tokenizers.rs

```diff
@@ -0,0 +1,52 @@
+use rust_tokenizers::{
+    tokenizer::{BertTokenizer, Tokenizer, TruncationStrategy},
+    Offset,
+};
+use std::ops::Range;
+
+use crate::{ChunkCapacity, ChunkSize, ChunkSizer};
+
+impl ChunkSizer for &BertTokenizer {
+    fn chunk_size(&self, chunk: &str, capacity: &impl ChunkCapacity) -> ChunkSize {
+        ChunkSize::from_offsets(
+            self.encode(
```

Comment:
I probably would have just used [tokenize_with_offsets](https://docs.rs/rust_tokenizers/8.1.1/rust_tokenizers/tokenizer/trait.Tokenizer.html#method.tokenize_with_offsets), but I guess this has the benefit that it might stop tokenization early? If so, this would be a performance win.

I am just not sure about setting the stride as capacity.start... but I might need to think about that some more... you may know more than me here.

Since you say there are potential issues where it isnt' chunking though, I would try the simpler method and see if it works. If it does, my guess is there is something wrong in these parameters that needs fixing

Reply:
I just took the first example from the page that had the required data.
When I switched to `tokenize_with_offsets` it works.